tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSGen.scala

```diff
@@ -147,27 +151,42 @@ private[emitter] final class JSGen(val semantics: Semantics,
   }
 
   def genRawJSClassConstructor(className: String)(
-      implicit globalKnowledge: GlobalKnowledge, pos: Position): Tree = {
+      implicit globalKnowledge: GlobalKnowledge,
+      pos: Position): WithGlobals[Tree] = {
+    genRawJSClassConstructor(className, keepAllGlobalVarNames = true)
+  }
+
+  def genRawJSClassConstructor(className: String,
+      keepAllGlobalVarNames: Boolean)(
```

Comment:
Consider calling the param `keepOnlyDangerousVarNames` for improved readability.